lib/logstash/outputs/s3/temporary_file.rb

```diff
@@ -61,10 +69,38 @@ def empty?
         def self.create_from_existing_file(file_path, temporary_folder)
           key_parts = Pathname.new(file_path).relative_path_from(temporary_folder).to_s.split(::File::SEPARATOR)
 
+          # recover gzip file and compress back before uploading to S3
+          if file_path.end_with?("." + GZIP_EXTENSION)
+            file_path = self.recover(file_path)
+          end
           TemporaryFile.new(key_parts.slice(1, key_parts.size).join("/"),
-                         ::File.open(file_path, "r"),
+                         ::File.exist?(file_path) ? ::File.open(file_path, "r") : nil, # for the nil case, file size will be 0 and upload will be ignored.
```

Comment:
Supplying a `nil` file descriptor here will cause the `size` method to raise rather than return `0`

Reply:
Oh sorry, missed to update the `TemporaryFile` class, now included.

Reply:
yeah, I double checked that forward-able methods will not be called when restoration fails. In the future, it would be nice if I can find a strategy to figure out recovery status rather than checking file existence.

Reply:
Is there a benefit to returning a `TemporaryFile` with a `nil` `fd` over returning `nil` here?

Reply:
Recovered file will not exist if recovery fails or restored txt file is empty. That leads to bring a failure (no file exist exception as I remember) when checking a size.